cmd/podman/machine/init.go

```diff
@@ -80,6 +80,14 @@ func initMachine(cmd *cobra.Command, args []string) error {
 	if len(args) > 0 {
 		initOpts.Name = args[0]
 	}
+	cfg := registry.PodmanConfig()
+
+	if cfg.Engine.MachineImage == "stable" || cfg.Engine.MachineImage == "testing" {
+		initOpts.ImageStream = cfg.Engine.MachineImage
+	} else if initOpts.ImagePath == "" {
+		initOpts.ImagePath = cfg.Engine.MachineImage
+	}
+
```

Comment:
I can't wrap my head around this conditional. Shouldn't `initOpts` (command-line options, I assume) override a config setting? And, I'm not sure all the fallback cases are covered here. But it's late in my day, so I'll need to look again tomorrow.

Reply:
ah yeah, this is a bit confusing. basically machine is get an image 2 ways: either pulling the default fcos image of which there are two options: `stable` and `testing`. machine can also take a path through `--image-path`. the `machine_image` in the config can do both, by setting it to `stable`, `testing`, or `/path/to/image`.
this conditional basically says, check if the config is set to a pull stream. if its, set the pull stream. if it's not, check if `imagePath` is set via a flag. if the flag is not set, this means that we may have a path in the config, so set the `imagePath` to what's specified in the config.

this is actually super confusing, and I was wondering if it would be just a better idea to allow --image-path to also take stable or testing, and maybe rename it to --image. it should make this logic more transparent.

Reply:
Why not just this:
```
flags.StringVar(&initOpts.ImagePath, ImagePathFlagName, cfg.Engine.MachineImage, "Path to qcow image")
```
and then check in vm.Init() if path == "stable" or "testing" and download fcos in this case otherwise assume its a url.
There is no need for the extra ImageStream option IMO.

Reply:
Sure, we could do that too @Luap99